### PR TITLE
155369 - Handling Scenario : Requesting s57 and Getting s63 Exchange Set (If 700+ Mb)

### PR DESCRIPTION
**Pull request by ratnas104142**
This PR consists changes of [PBI 155369](https://dev.azure.com/ukhydro/Exchange%20Set%20Service/_workitems/edit/155369) -
- Code changes for Handling Scenario : Requesting s57 and Getting s63 Exchange Set (If data is 700+ Mb)
- Unit tests

### CODE REVIEW

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs [hunk 1]
```diff
@@ -257,18 +256,54 @@ public async Task WhenLargeExchangeSetRequested_ThenPostProductIdentifiersReturn
...
                  .Returns(exchangeSetServiceResponse);
 
-            var result = (BadRequestObjectResult)await controller.PostProductIdentifiers(productIdentifiers, callbackUri, exchangeSetStandard.ToString());
+            var result = (BadRequestObjectResult)await controller.PostProductIdentifiers(productIdentifiers, callbackUri, ExchangeSetStandard.s57.ToString());
             var errors = (ErrorDescription)result.Value;
 
-            Assert.AreEqual(400, result.StatusCode);
-            Assert.AreEqual("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.", errors.Errors.Single().Description);
+            result.StatusCode.Should().Be(400);
+            errors.Errors.Single().Description.Should().BeEquivalentTo("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```
Comment (Nadeem14456) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs:263]:
```suggestion
            errors.Errors.Single().Description.Should().Be("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs [hunk 2]
```diff
@@ -441,18 +474,52 @@ public async Task WhenLargeExchangeSetRequested_ThenPostProductDataByProductVers
...
 
             var result = (BadRequestObjectResult)await controller.PostProductDataByProductVersions(new List<ProductVersionRequest>()
-                            { new ProductVersionRequest() { ProductName = "demo" } }, "", exchangeSetStandard.ToString());
+                            { new() { ProductName = "demo" } }, "", ExchangeSetStandard.s57.ToString());
             var errors = (ErrorDescription)result.Value;
 
-            Assert.AreEqual(400, result.StatusCode);
-            Assert.AreEqual("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.", errors.Errors.Single().Description);
+            result.StatusCode.Should().Be(400);
+            errors.Errors.Single().Description.Should().BeEquivalentTo("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```
Comment (Nadeem14456) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs:481]:
```suggestion
            errors.Errors.Single().Description.Should().Be("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```

File: UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs [hunk 3]
```diff
@@ -588,18 +653,51 @@ public async Task WhenLargeExchangeSetRequested_ThenGetProductDataSinceDateTimeR
...
                 .Returns(exchangeSetServiceResponse);
 
-            var result = (BadRequestObjectResult)await controller.GetProductDataSinceDateTime("Wed, 21 Oct 2015 07:28:00 GMT", "https://www.abc.com", exchangeSetStandard.ToString());
+            var result = (BadRequestObjectResult)await controller.GetProductDataSinceDateTime("Wed, 21 Oct 2015 07:28:00 GMT", "https://www.abc.com", ExchangeSetStandard.s57.ToString());
             var errors = (ErrorDescription)result.Value;
 
-            Assert.AreEqual(400, result.StatusCode);
-            Assert.AreEqual("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.", errors.Errors.Single().Description);
+            result.StatusCode.Should().Be(400);
+            errors.Errors.Single().Description.Should().BeEquivalentTo("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```
Comment (Nadeem14456) [inline UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.API.UnitTests/Controllers/ProductDataControllerTests.cs:660]:
```suggestion
            errors.Errors.Single().Description.Should().Be("The Exchange Set requested is very large and will not be created, please use a standard Exchange Set provided by the UKHO.");
```

**Review by Nadeem14456 — CHANGES_REQUESTED** [review]
some minor comments

**Review by shirinbt14926 — APPROVED** [review]
LGTM

**Review by DanOwenUKHO — APPROVED** [review]

**Pull request merged**